components/script/dom/webglrenderingcontext.rs

```diff
@@ -435,9 +435,14 @@ impl WebGLRenderingContext {
     }
 
     fn mark_as_dirty(&self) {
-        self.canvas
-            .upcast::<Node>()
-            .dirty(NodeDamage::OtherNodeDamage);
+        // If we don't have a bound framebuffer, then don't mark the canvas
+        // as dirty.
+        match self.bound_framebuffer.get() {
```

Comment:
We want to invest this; only mark as dirty when there is no bound framebuffer. We can use is_none() instead of matching for this.